### ISSUE TITLE
Able to write `LineString` and `Polygon` geometry,

### ISSUE THREAD

**Issue by ErickChacon** [JuliaGeo/Shapefile.jl#89]
I started a PR in GeoTables (JuliaEarth/GeoTables.jl#15) to be able to write `Meshes.jl` geometries to `.shp` using `Shapefile.jl`. However, we have problems when trying to write non-multi geometries like (`LineString` and `Polygon`).

```
ERROR: KeyError: key GeoInterface.LineStringTrait not found
Stacktrace:
 [1] getindex
   @ ./dict.jl:498 [inlined]
 [2] write(path::String, o::Shapefile.Writer; force::Bool)
   @ Shapefile ~/.julia/packages/Shapefile/TRwfU/src/writer.jl:140
 [3] #write#36
   @ ~/.julia/packages/Shapefile/TRwfU/src/writer.jl:266 [inlined]
 [4] save(fname::String, geotable::GeoTables.GeoTable{GeoJSON.FeatureCollection{2, Float64}}; kwargs::Base.Pairs{Symbol, Bool, Tuple{Symbol}, NamedTuple{(:force,), Tuple{Bool}}})
   @ GeoTables ~/documents/repositories/GeoTables.jl/src/GeoTables.jl:64
 [5] top-level scope
   @ REPL[1965]:1



ERROR: KeyError: key GeoInterface.PolygonTrait not found
Stacktrace:
 [1] getindex
   @ ./dict.jl:498 [inlined]
 [2] write(path::String, o::Shapefile.Writer; force::Bool)
   @ Shapefile ~/.julia/packages/Shapefile/TRwfU/src/writer.jl:140
 [3] #write#36
   @ ~/.julia/packages/Shapefile/TRwfU/src/writer.jl:266 [inlined]
 [4] save(fname::String, geotable::GeoTables.GeoTable{GeoJSON.FeatureCollection{2, Float64}}; kwargs::Base.Pairs{Symbol, Bool, Tuple{Symbol}, NamedTuple{(:force,), Tuple{Bool}}})
   @ GeoTables ~/documents/repositories/GeoTables.jl/src/GeoTables.jl:64
 [5] top-level scope
   @ REPL[1960]:1
```

We can solve this by transforming the `StringLine` and `Polygon` to `MultiStringLine` and `MultiPolygon`; however this will required to do by every package that wants to use `Shapefile.jl` to write to disk. Can that issue be solved here, so other packages using `GeoInterface.jl` integration can also benefit from it?

**Comment by rafaqz** [comment]
The reason is that the shapefile format doesn't have Polygon or LineString, so there is no direct mapping.
Just something they call a polygon that's actually a  multipolygon, with holes mixed randomly with exteriors. Its wonderful.

Either you can just wrap in MultiPolygon before writing, or you can PR here so the writer accepts polygons and LinearRing and writes them as MultiPolygonTrait. 

`GeoInterface.getring` will give what you need to write for both Polygon and MultiPolygon, but you may need to special case LinearRing as it is itself the ring you need to write.

**Comment by ErickChacon** [comment]
@rafaqz Thank you for your explanation. I will see if I can help here, the idea would be to convert to `Polyline` and `MultiPolygon` before writing.

With respect to reading no problem with obtaining `MultiPolygon`.

**Comment by rafaqz** [comment]
GeoInterface has wrapper types for things like this now - MultiPolygon will wrap a vector of any PolygonTrait object, for example.

**Comment by rafaqz** [comment]
I it would be good to fix this here rather than special casing packages.

**Comment by ErickChacon** [comment]
> I it would be good to fix this here rather than special casing packages.

Yes, I am already working on this. I have this working now, I just need to check test are passing.
